services/shuttle-poem/src/lib.rs

```diff
@@ -21,14 +21,23 @@
 /// A wrapper type for [poem::Endpoint] so we can implement [shuttle_runtime::Service] for it.
 pub struct PoemService<T>(pub T);
 
+#[poem::handler]
+fn healthz() -> poem::http::StatusCode {
+    poem::http::StatusCode::OK
+}
+
 #[shuttle_runtime::async_trait]
 impl<T> shuttle_runtime::Service for PoemService<T>
 where
     T: poem::Endpoint + Send + 'static,
 {
     async fn bind(mut self, addr: std::net::SocketAddr) -> Result<(), shuttle_runtime::Error> {
+        let app = poem::Route::new()
+            .at("/healthz", poem::get(healthz))
+            .nest("/", self.0);
```

Comment:
Maybe we should swap this so `/healthz` is the outermost route?

Reply:
If you mean doing this
```rust
self.0 = self.0.at("/healthz", poem::get(healthz));
```
we can't do this because the trait bound on `PoemService` expects the trait [`poem::Endpoint`](https://docs.rs/poem/1.3.55/poem/trait.Endpoint.html) which doesn't have a why to chain requests so we need to create our own `poem::Route` or change the trait bound.

Reply:
Hmm, maybe @oddgrd  referred to something like this? Haven't tried it myself so not sure if it works.
```suggestion
        let app = poem::Route::new()
            .at("/", self.0),
            .at("/healthz", poem::get(healthz));
```

Reply:
that works, converted to this in the latest commit